uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js

```diff
@@ -11,6 +11,71 @@ define([
   instances.factory('LeaveRequestInstance', ['$q', 'checkPermissions', 'OptionGroup',
     'shared-settings', 'ModelInstance', 'LeaveRequestAPI',
     function ($q, checkPermissions, OptionGroup, sharedSettings, ModelInstance, LeaveRequestAPI) {
+      /**
+       * Amends balance change breakdown if calculation unit absence type is "hours".
+       * It changes the balance first and last days (only first day in case of single day request)
+       *   by setting the selected deductions.
+       * @NOTE this function mutates the "balanceChange" object
+       *
+       * @param  {Object} balanceChange
+       * @param  {String} calculationUnit
+       * @return {Promise} resolve with amended balance change in case of "hours" absence type
+       *   or returns unamended balance change in case of "days" absence type
+       */
+      function amendHourlyBalanceChangeBreakdown (balanceChange, calculationUnit) {
+        var breakdown = balanceChange.breakdown;
+        var request = this;
+
+        if (calculationUnit !== 'hours') {
+          return balanceChange;
+        }
+
+        return OptionGroup.valuesOf('hrleaveandabsences_leave_request_day_type')
+          .then(function (dayTypes) {
+            if (balanceChange) {
+              amendHourlyBalanceChangeForDay.call(request, _.first(_.values(breakdown)), 'from', dayTypes);
+
+              if (breakdown.length > 1) {
+                amendHourlyBalanceChangeForDay.call(request, _.last(_.values(breakdown)), 'to', dayTypes);
+              }
+
+              amendHourlyBalanceChangeAmount(balanceChange);
+            }
+
+            return balanceChange;
+          });
+      }
+
+      /**
+       * Amends balance change amount if calculation unit absence type is "hours".
+       * @NOTE this function mutates the "balanceChange" object
+       *
+       * @param {Object} balanceChange
+       */
+      function amendHourlyBalanceChangeAmount (balanceChange) {
+        balanceChange.amount = _.reduce(balanceChange.breakdown, function (updatedChange, day) {
+          return updatedChange - day.amount;
+        }, 0);
+      }
+
+      /**
+       * Amends a particular day balance change if calculation unit absence type is "hours".
```

Comment:
This function does not check ` if calculation unit absence type is "hours"` so please remove that part.

Reply:
Done